numpy/core/src/multiarray/methods.c

```diff
@@ -1905,11 +2029,26 @@ PyArray_Dumps(PyObject *self, int protocol)
     if (protocol < 0) {
         protocol = 2;
     }
+
+    if (protocol == 5){
+#if PY_VERSION_HEX >=0x03080000
+        cpick = PyImport_ImportModule("pickle");
+#elif PY_VERSION_HEX < 0x03080000 && PY_VERSION_HEX >= 0x03060000
+        cpick = PyImport_ImportModule("pickle5");
+#else
+        return NULL;
```

Comment:
We cannot do this, without setting a proper error message.

For instance with Python 3.5 I get the following with the current implementation:

```python
>>> import numpy as np                                                                                                                                                          
>>> a = np.arange(12)                                                                                                                                                           
>>> s = a.dumps(protocol=6)                                                                                                                                                     
Traceback (most recent call last):
  File "<ipython-input-9-2e271e7e7bc8>", line 1, in <module>
    s = a.dumps(protocol=6)
ValueError: pickle protocol must be <= 4

>>> s = a.dumps(protocol=5)                                                                                                                                                     
Traceback (most recent call last):
  File "<ipython-input-10-7689e2059d65>", line 1, in <module>
    s = a.dumps(protocol=5)
SystemError: <built-in method dumps of numpy.ndarray object at 0x7f2a181c22b0> returned NULL without setting an error
```

For Python <= 3.6, we need to always get `ValueError: pickle protocol must be <= 4`.